integration-tests/contract-security-on-chain-views.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { Protocols } from '@taquito/taquito';
+import { CONFIGS } from './config';
+
+/**   
+ * This "naive*" testcase assumes that the on-chain view stack is not correctly separated from the caller's stack. 
+ * Thus, this testcase assumes that there is only one stack, whereas the elements of the on-chain stack are on top. 
+ * Naive - meaning: WE just try it without thinking whether this test makes sense in regards with the used underlying architecture. 
+ * We think of the underlying architecture (type system, stack separation, etc.) as a black box.
+ * 
+ * TC-V-012: On-chain view - add instruction
+ * TC-V-013: On-chain view - dig instruction
+ * TC-V-014: On-chain view - dup instruction
+**/
+
+CONFIGS().forEach(({ lib, rpc, setup, protocol }) => {
+  const Tezos = lib;
+  const mondaynet = protocol === Protocols.ProtoALpha ? test: test.skip;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    mondaynet("Verify we can access the stack of the caller by using the instruction add.", async (done) => {
+      try {
+        const opGetter = await Tezos.contract.originate({
+          code: ` { parameter address;
+            storage nat;
+            code
+              {
+                UNPAIR;
+                UNIT;
+                VIEW "rogue" nat;
+                IF_NONE { FAIL } {};
+                ADD;
+                NIL operation;
+                PAIR;
+              };}`,
+          init: `0`
+        });
+
+        await opGetter.confirmation();
+        expect(opGetter.hash).toBeDefined();
+
+      } catch (error: any) {
```

Comment:
The test doesn't seem to throw an error. 
I am wondering why this error `expect(error.message).toContain('michelson_v1.invalid_expression_kind');` is expected

Reply:
I fixed this test so that it gets the expected "bad stack" error. the problem was the init was wrong.